src/dependency_groups/_implementation.py

```diff
@@ -134,9 +132,9 @@ def _parse_group(
 
         raw_group = self.dependency_groups[group]
         if not isinstance(raw_group, list):
-            raise ValueError(f"Dependency group '{group}' is not a list")
+            raise TypeError(f"Dependency group '{group}' is not a list")
```

Comment:
Tests need to be updated with this change if it looks good to you.

Reply:
:+1: Yes, IMO this is just a bug. Good catch!